graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphQLService.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.graphql;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import org.dataloader.DataLoaderRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.QueryParams;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionInput.Builder;
+import graphql.ExecutionResult;
+import graphql.ExecutionResultImpl;
+import graphql.GraphQL;
+import graphql.GraphqlErrorException;
+import graphql.com.google.common.collect.ImmutableMap;
+
+final class DefaultGraphQLService extends AbstractHttpService implements GraphQLService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultGraphQLService.class);
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```

Comment:
I guess I might have to use [JsonUtil](https://github.com/line/armeria/pull/3458/files#diff-a5eff28dc9e5a97a9f1c293b5988c5f7c6287537a8e459c40bca080f6e6a6282R27) here. 🤔